specs/recording-requests.md

```diff
@@ -0,0 +1,21 @@
+# Recording requests
+
+Recording requests allows Developer Proxy to capture requests and related messages that were intercepted by the proxy while in record mode. The captured requests are passed for further processing to the Developer Proxy plugins. This feature is foundational for scenarios such as permission analysis, building tailored SDKs, audit reports, etc.
+
+## History
+
+| Version | Date | Comments | Author |
+| ------- | ---- | -------- | ------ |
+| 1.0 | 2023-02-01 | Initial specifications | @waldekmastykarz |
+
+## Implementation
+
+Developer Proxy can be put in recording mode either by using the `--record` command line option or by pressing `r` after starting the proxy. Recording can be stopped by pressing `s` while running the proxy or by closing the proxy using `CTRL+C`. When the recording is stopped while proxy is running, users can start a new recording session by pressing `r` again.
+
+While recording, Developer Proxy will show and indicator in the terminal window. The indicator will be displayed in the top right corner of the window and will be hidden if the proxy is not recording.
+
+Proxy is recording requests by subscribing to a `LogRequest` event which is raised each time proxy logs a request-related log message. The event contains information about the logged message.
```

Comment:
Is this event to be raised on each call to `ILoger.LogRequest`?
Does it happen before the logging happens or after?

Reply:
Yes, it's this event. I suggest we raise the event after the message was logged. It aligns more with the "Logged" aspect (past tense).

Reply:
Perhaps a slight tweak to the event name `RequestLogged` gets this to the place we want it.

IMO, for the moment we leave it and address when/if we focus on custom loggers.